daft/runners/ray_runner.py

```diff
@@ -515,13 +519,16 @@ def _build_partitions(task: PartitionTask[ray.ObjectRef]) -> list[ray.ObjectRef]
 
     if isinstance(task.instructions[0], ReduceInstruction):
         build_remote = reduce_and_fanout if isinstance(task.instructions[-1], FanoutInstruction) else reduce_pipeline
+        build_remote = build_remote.options(**ray_options)
+        [metadatas_ref, *partitions] = build_remote.remote(task.instructions, task.inputs)
```

Comment:
what do you think about doing a `ray.wait` on all of the object_refs in `task.inputs` before issuing out the remote fn.

Reply:
I was thinking since we have to wait for all the map-fanouts being complete anyways before the first reduce task can even start. Scheduling the reduce tasks might outcompete the fan-outs and be blocked on waiting for the map phase. This way we ensure the fan outs finish as fast as possible.